src/grafana_agent.py

```diff
@@ -262,7 +262,7 @@ def _update_status(self) -> bool:
         Returns:
             True if the status was set to Active; False otherwise.
         """
-        if len(self.model.relations["metrics-endpoint"]):
+        if self.scrape_relation_name and len(self.model.relations[self.scrape_relation_name]):
             if not len(self.model.relations[REMOTE_WRITE_RELATION_NAME]):
                 self.unit.status = WaitingStatus("no related Prometheus remote-write")
                 return False
```

Comment:
I'm wondering whether we should even have this block? We don't do this with loki_push_api right?

Also, if we keep it, it should be `BlockedStatus`

Reply:
it's weird that we have this block for prometheus but not loki.

Besides we are using a `common_exit_hook`ish pattern with `update_config` method where we set the status, but in `_update_status` we override status.

I would continue this on a separate PR.

Reply:
Let's follow this here: https://github.com/canonical/grafana-agent-k8s-operator/issues/104